src/main/java/net/fabricmc/loom/task/RemapTaskConfiguration.java

```diff
@@ -69,32 +67,37 @@ public static void setupRemap(Project project) {
 
 			// Setup the input file and the nested deps
 			task.getInputFile().convention(jarTask.getArchiveFile());
+			task.dependsOn(tasks.named(JavaPlugin.JAR_TASK_NAME));
 			Configuration includeConfiguration = project.getConfigurations().getByName(Constants.Configurations.INCLUDE);
 			task.getNestedJars().from(new IncludedJarFactory(project).getNestedJars(includeConfiguration));
 		});
 
 		// Configure the default jar task
 		tasks.named(JavaPlugin.JAR_TASK_NAME, AbstractArchiveTask.class).configure(task -> {
 			task.getArchiveClassifier().convention("dev");
-			task.finalizedBy(remapJarTaskProvider);
 			task.getDestinationDirectory().set(new File(project.getBuildDir(), "devlibs"));
 		});
 
+		tasks.named("assemble").configure(task -> task.dependsOn(remapJarTaskProvider));
```

Comment:
FYI there's a constant in `BasePlugin` for this

Reply:
Ah thanks, I didnt know where it was hiding :D

Reply:
Also it might be nice to hook `remapSourcesJar` to this as well